src/Datagrid/Pager.php

```diff
@@ -27,15 +27,71 @@ class Pager extends BasePager
 {
     protected $queryBuilder = null;
 
-    public function computeNbResult()
+    /**
+     * @var int
+     */
+    private $resultsCount = 0;
+
+    public function countResults(): int
     {
-        $countQuery = clone $this->getQuery();
+        // NEXT_MAJOR: just return "$this->resultsCount" directly.
+        $deprecatedCount = $this->getNbResults('sonata_deprecation_mute');
 
-        if (\count($this->getParameters()) > 0) {
-            $countQuery->setParameters($this->getParameters());
+        if ($deprecatedCount === $this->resultsCount) {
+            return $this->resultsCount;
+        }
+
+        @trigger_error(sprintf(
+            'Relying on the protected property "%s::$nbResults" and its getter/setter is deprecated since sonata-project/doctrine-mongodb-admin-bundle 3.x and will fail 4.0. Use "countResults()" and "setResultsCount()" instead.',
+            self::class,
+        ), E_USER_DEPRECATED);
+
+        return $deprecatedCount;
+    }
+
+    public function setResultsCount(int $count): void
```

Comment:
This method should be private IMHO.

Reply:
We also merged it as public here though: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1257/files#diff-f13ffebef517d741bc6f655c9465d29a5154a3c9a1ac5746acb85654dfc606f4R92

Reply:
I'll do the change in my PR: https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1248